hedera-node/src/main/java/com/hedera/services/fees/calculation/contract/queries/ContractCallLocalResourceUsage.java

```diff
@@ -222,7 +221,7 @@ private ProcessableBlockHeader stubbedBlockHeader(long timestamp) {
 				Address.ZERO, //Coinbase might be the 0.98 address?
 				Difficulty.ONE,
 				0,
-				12_500_000L,
+				properties.maxGas(),
```

Comment:
This is the gas limit per block and it should be `properties.maxGas` however we still have to set the `gasLimit` of the actual transaction to the `properties.maxGas` otherwise we will:
- limit the tx to hardcoded `long gasLimit = 1_000_000L; ` on line 160
- allow for `maxGas` in the block -> effectively limiting the transaction to 1M gas limit.

On line 160 where you need to change it to `maxGas`, you need to verify that the gas coming from the grpc operation is not `>` than the `maxGas`